api/src/main/java/bio/terra/tanagra/api/QuerysService.java

```diff
@@ -208,6 +229,97 @@ public List<EntityInstanceCount> runInstanceCountsQuery(
+          "Only CRITERIA_OCCURENCE entity groups support display hints queries");
+    }
+    CriteriaOccurrence criteriaOccurrence = (CriteriaOccurrence) entityGroup;
+    AuxiliaryDataMapping auxDataMapping =
+        criteriaOccurrence.getModifierAuxiliaryData().getMapping(mappingType);
+
+    TableVariable primaryTableVar = TableVariable.forPrimary(auxDataMapping.getTablePointer());
+    List<TableVariable> tableVars = Lists.newArrayList(primaryTableVar);
+
+    List<FieldVariable> selectFieldVars =
+        auxDataMapping.getFieldPointers().entrySet().stream()
+            .map(
+                entry -> entry.getValue().buildVariable(primaryTableVar, tableVars, entry.getKey()))
+            .collect(Collectors.toList());
+    // TODO: Centralize this schema definition used both here and in the indexing job. Also handle
+    // other enum_value data types.
+    List<ColumnSchema> columnSchemas =
+        List.of(
+            new ColumnSchema(MODIFIER_AUX_DATA_ID_COL, CellValue.SQLDataType.INT64),
+            new ColumnSchema(MODIFIER_AUX_DATA_ATTR_COL, CellValue.SQLDataType.STRING),
+            new ColumnSchema(MODIFIER_AUX_DATA_MIN_COL, CellValue.SQLDataType.FLOAT),
+            new ColumnSchema(MODIFIER_AUX_DATA_MAX_COL, CellValue.SQLDataType.FLOAT),
+            new ColumnSchema(MODIFIER_AUX_DATA_ENUM_VAL_COL, CellValue.SQLDataType.INT64),
+            new ColumnSchema(MODIFIER_AUX_DATA_ENUM_DISPLAY_COL, CellValue.SQLDataType.STRING),
+            new ColumnSchema(MODIFIER_AUX_DATA_ENUM_COUNT_COL, CellValue.SQLDataType.INT64));
+
+    // Build the WHERE filter variables from the entity filter.
+    FieldVariable entityIdFieldVar =
+        selectFieldVars.stream()
+            .filter(fv -> fv.getAlias().equals(MODIFIER_AUX_DATA_ID_COL))
+            .findFirst()
+            .get();
+    BinaryFilterVariable filterVar =
+        new BinaryFilterVariable(
+            entityIdFieldVar, BinaryFilterVariable.BinaryOperator.EQUALS, relatedEntityId);
+
+    Query query =
+        new Query.Builder().select(selectFieldVars).tables(tableVars).where(filterVar).build();
+    LOGGER.info("Generated query: {}", query.renderSQL());
+
+    return new QueryRequest(query.renderSQL(), new ColumnHeaderSchema(columnSchemas));
+  }
+
+  public Map<String, DisplayHint> runDisplayHintsQuery(
+      DataPointer dataPointer, QueryRequest queryRequest) {
+    QueryResult queryResult = dataPointer.getQueryExecutor().execute(queryRequest);
+
+    Map<String, DisplayHint> displayHints = new HashMap<>();
+    Map<String, List<EnumVal>> runningEnumValsMap = new HashMap<>();
+    Iterator<RowResult> rowResultsItr = queryResult.getRowResults().iterator();
+    while (rowResultsItr.hasNext()) {
+      RowResult rowResult = rowResultsItr.next();
+      String attrName = rowResult.get(MODIFIER_AUX_DATA_ATTR_COL).getLiteral().getStringVal();
+
+      OptionalDouble min = rowResult.get(MODIFIER_AUX_DATA_MIN_COL).getDouble();
+      if (min.isPresent()) {
```

Comment:
Each row in the `MODIFIER_AUX_DATA` table applies to a single attribute. Each attribute can only have one kind of hint. We can have multiple different hints for a single entity, but it's at most one per attribute.

In your example for `measurement` instance=blood pressure, there could be 3 attributes each with a display hint:
- `systolic` attribute has a numeric range 70-90
- `diastolic` attribute has a numeric range 110-130
- `category` attribute has enum values low/normal/high

Reply:
Hmm, apparently I don't understand what the data looks like. Starting with a simpler example like BMI, my understanding was that BMI is an instance of the measurement entity. Measurements have two attributes, a numeric value and a descriptive value (plus name etc.). To calculate the display hints, the set of measurement_occurrences where the measurement_id equals "BMI" would be queried. From there, the minimum value, the maximum value, and the set of unique descriptive values along with their counts would be collected. So each instance has a set of display hints.

From the example, I don't understand how blood pressure has unique attributes. I thought those would be the same across all measurement instances. It was also a bad example, I suppose, because it's probably actually a branch node in the hierarchy and systolic and diastolic are children and leaf nodes. The two leaf nodes would each have a a display hint associated with them containing a min, max, and enum values.

Where am I getting off track?

Reply:
I think we're on the same page about how display hints map to attributes, but maybe I'm not being clear about how this information is generated and stored in the index tables.

The `measurement_occurrence` entity has 2 attributes related to value: `value_numeric` (simple) and `value_enum` (key_display). Each attribute can have at most 1 display hint. For display hints filtered through the related `measurement` instance BMI, that could look like:
- `value_numeric` attribute has a numeric range of 1-100
- `value_enum` attribute has enum values low/normal/high
So there are 2 `measurement_occurrence` attribute display hints for the single related `measurement` instance.

This information is stored in the index tables across multiple rows. e.g. Here's the display hint rows for "Body height", which is a `measurement` instance with 2 types of hints.
![Screen Shot 2022-11-03 at 1 42 16 PM](https://user-images.githubusercontent.com/12446128/199795545-6c47f420-1061-4652-8fe5-1e4118a9ec5f.png)

This part of the code is iterating through the rows in the index table, not `measurement` ids. Each index table row contains information about one enum value or a numeric range, not both.

Reply:
I see. That makes sense.